cmd/server-main.go

```diff
@@ -101,115 +98,24 @@ EXAMPLES:
-	port int    // Will be valid for distributed XL
-	path string // Will be valid for all configs
-}
-
-// Returns string form.
-func (ep storageEndPoint) String() string {
-	var str []string
-	if ep.host != "" {
-		str = append(str, ep.host)
-	}
-	if ep.port != 0 {
-		str = append(str, strconv.Itoa(ep.port))
-	}
-	if ep.path != "" {
-		str = append(str, ep.path)
-	}
-	return strings.Join(str, ":")
-}
-
-// Returns if ep is present in the eps list.
-func (ep storageEndPoint) presentIn(eps []storageEndPoint) bool {
-	for _, entry := range eps {
-		if entry == ep {
-			return true
-		}
-	}
-	return false
-}
-
-// Parse end-point (of the form host:port:path or host:path or path)
-func parseStorageEndPoint(ep string, defaultPort int) (storageEndPoint, error) {
-	if runtime.GOOS == "windows" {
-		// Try to match path, ex. C:\export or export
-		matched, err := regexp.MatchString(`^([a-zA-Z]:\\[^:]+|[^:]+)$`, ep)
-		if err != nil {
-			return storageEndPoint{}, err
-		}
-		if matched {
-			return storageEndPoint{path: ep}, nil
-		}
-
-		// Try to match host:path ex. 127.0.0.1:C:\export
-		re, err := regexp.Compile(`^([^:]+):([a-zA-Z]:\\[^:]+)$`)
-		if err != nil {
-			return storageEndPoint{}, err
-		}
-		result := re.FindStringSubmatch(ep)
-		if len(result) != 0 {
-			return storageEndPoint{host: result[1], port: defaultPort, path: result[2]}, nil
-		}
-
-		// Try to match host:port:path ex. 127.0.0.1:443:C:\export
-		re, err = regexp.Compile(`^([^:]+):([0-9]+):([a-zA-Z]:\\[^:]+)$`)
-		if err != nil {
-			return storageEndPoint{}, err
-		}
-		result = re.FindStringSubmatch(ep)
-		if len(result) != 0 {
-			portInt, err := strconv.Atoi(result[2])
-			if err != nil {
-				return storageEndPoint{}, err
-			}
-			return storageEndPoint{host: result[1], port: portInt, path: result[3]}, nil
-		}
-		return storageEndPoint{}, errors.New("Unable to parse endpoint " + ep)
-	}
-	// For *nix OSes
-	parts := strings.Split(ep, ":")
-	var parsedep storageEndPoint
-	switch len(parts) {
-	case 1:
-		parsedep = storageEndPoint{path: parts[0]}
-	case 2:
-		parsedep = storageEndPoint{host: parts[0], port: defaultPort, path: parts[1]}
-	case 3:
-		port, err := strconv.Atoi(parts[1])
-		if err != nil {
-			return storageEndPoint{}, err
-		}
-		parsedep = storageEndPoint{host: parts[0], port: port, path: parts[2]}
-	default:
-		return storageEndPoint{}, errors.New("Unable to parse " + ep)
-	}
-	return parsedep, nil
-}
-
-// Parse an array of end-points (passed on the command line)
-func parseStorageEndPoints(eps []string, defaultPort int) (endpoints []storageEndPoint, err error) {
+// Parse an array of end-points (from the command line)
+func parseStorageEndpoints(eps []string) (endpoints []*url.URL, err error) {
 	for _, ep := range eps {
 		if ep == "" {
-			continue
+			return nil, errInvalidArgument
 		}
-		var endpoint storageEndPoint
-		endpoint, err = parseStorageEndPoint(ep, defaultPort)
+		var u *url.URL
+		u, err = url.Parse(ep)
```

Comment:
does url.Parse support parsing http://host:9000/D:/export ? if not does the syntax look for windows?